aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -129,12 +132,15 @@ module aptos_framework::transaction_validation {
             );
         };
 
+        let apt_fa = coin::apt_fa_metadata(); // ensure_paired_metadata<AptosCoin>();
+
         let max_transaction_fee = txn_gas_price * txn_max_gas_units;
         assert!(
-            coin::is_account_registered<AptosCoin>(gas_payer),
+            primary_fungible_store::primary_store_exists(gas_payer, apt_fa),
             error::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT),
         );
```

Comment:
This line is unnecessary because if it does not exist, line 142 will return 0 as balance.